test/modules/standard/Spawn/spawn-kill.chpl

```diff
@@ -0,0 +1,9 @@
+use Spawn;
+use Time;
+
+var sub = spawn(["sleep", "10"]);
+sleep(1);
+sub.kill();
+sub.poll();
+assert(sub.running == false);
+assert(sub.exit_status != 0);
```

Comment:
I think this should check that exit_status == -SIGKILL
Also, I'd increase the timeout. You're not going to wait all 10 seconds in a normal run, but increasing it to 100 might add some robustness to the test. It's arbitrary in any case...